server/src/main/java/org/opensearch/gateway/ReplicaShardAllocator.java

```diff
@@ -441,9 +470,19 @@ private MatchingNodes findMatchingNodes(
         return new MatchingNodes(matchingNodes, nodeDecisions);
     }
 
+    private Map<DiscoveryNode, StoreFilesMetadata> adaptToNodeShardStores(AsyncShardFetch.FetchResult<NodeStoreFilesMetadata> data) {
+        assert data.hasData();
```

Comment:
Will this condition always hold true ? Why do we need this assertion here?

Reply:
This was because we were doing early exits if shard did not have any data. Anyway the behavior has change in latest commits.